torchtune/models/clip/_position_embeddings.py

```diff
@@ -100,23 +100,314 @@ def __init__(
 
         self.gate = nn.Parameter(torch.zeros(1))
 
+        self._register_load_state_dict_pre_hook(self._load_state_dict_hook)
+
+    def _load_state_dict_hook(
+        self,
+        state_dict: Dict[str, Any],
+        prefix: str,
+        *args: Tuple[Any],
+        **kwargs: Dict[str, Any],
+    ) -> None:
+        """
+        Interpolates positional embeddings to accomodate different number of tiles
+        and tokens per tile, in case the model was instantiated with different
+        settings than the one you are loading the state dict from.
+
+        For more info, please check self._resize_local_position_embedding and
+        self._resize_global_position_embedding functions.
+
+        Args:
+            state_dict (Dict[str, Any]): The state dict to load.
+            prefix (str): The prefix of the state dict.
+            *args (Tuple[Any]): Additional positional arguments.
+            **kwargs (Dict[str, Any]): Additional keyword arguments.
+
+        Raises:
+            ValueError: if loaded local or global embedding n_tokens_per_tile is not derived
+                from a squared grid.
+            ValueError: if after interpolation, the shape of the loaded local embedding
+                is not compatible with the current embedding.
+            ValueError: if after interpolation, the shape of the loaded global embedding
+                is not compatible with the current embedding.
+        """
+        # local
```

Comment:
remove (unless you meant to add something else here)

Reply:
just trying to help the reader understand that this is the local portion, and later there is the global portion. But maybe not useful at all. Will remove